source/_posts/angular-session-storage.md

```diff
@@ -0,0 +1,299 @@
+
+### Session Storage
+
+Session storage is similar to cookies in that data is only stored for the duration of the current session. When a user closes their browser or a tab, the web app clears the data and deletes any previously stored information. Session storage is useful for storing sensitive data, such as login credentials.
+
+Session storage is an excellent way to improve the performance of your web applications by reducing the amount of data transferred between the client and server. It can also be used to store data more securely because the data is not stored in cookies where third-party sites can access it.
+
+### Cookies
+
+Cookies are small text files that are stored on a user's device by a website they visit. They are generated by the website's server and sent to the user's browser, where they are stored. Cookies store information about the user's interactions with the website, such as login credentials, user preferences, and shopping cart contents. Cookies are used for a variety of purposes, including authentication, personalization, tracking, and analytics.
+
+## Angular Session Storage
+
+Let's look at how to store, get, delete specific data, or remove all data in Angular's `sessionStorage`.
+
+API methods for working with key/value pair data are incorporated into session storage.
+
+It should be noted that both key and value are string types, and if you want to keep a different data type, you must convert it to a string first.
+
+### Saving data using sessionStorage
+
+Create a `dataSave()` function in the `app.component.ts` file to store data in `sessionStorage`.
+
+```typescript
+dataSave(){
+}
+```
+    
+Use `setItem` within the `dataSave()` function to keep the name in the `sessionStorage`.
+
+Syntax:
+
+```typescript
+sessionStorage.setItem('key', 'value');
+```
+
+Thus, this is how our `app.component.ts` will appear:
+
+```typescript
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  title = 'new_project';
+
+  dataSave(){
+    sessionStorage.setItem('name', 'Sanjeev');
+  }
+}
+```
+
+Now, open the `app.component.html` file and add a button to the `dataSave()` function with a click event. As a result, the `app.component.html` file will look something like this.
+
+```html
+<h1>Angular sessionStorage</h1>
+<button (click)="dataSave()" > Save data to sessionStorage</button>
+```
+
+{% img "save-data.jpg" "Save data in angular session storage" "lazy" %}
+
+When you click the "Save data to sessionStorage" button, the key "name" and the value "Sanjeev" will be saved.
+
+This can be verified by inspecting the page and going to `application > session storage`.
+
+{% img "session-storage.jpg" "Verify saved data in angular session storage" "lazy" %}
+
+If we close the browser or this tab, the session data stored will be automatically cleared.
+
+### Retrieve data from session storage

```

Comment:
It must be consistent. If you decided to use uppercase in headlines you should do it here as well. We talked about it earlier. It should be:
`Retrieve data from Session Storage`